### PR TITLE
add dynamic width to input

### PR DESCRIPTION
**Pull request by nirhart**

### CODE REVIEW

File: src/MultiSelect/InputWithTags.js [hunk 1]
```diff
@@ -31,10 +33,18 @@ class InputWithTags extends React.Component {
 
         {tags.map(({label, ...rest}) => <Tag key={rest.id} onRemove={onRemoveTag} {...rest}>{label}</Tag>)}
         <span className={styles.input} data-hook="inner-input-with-tags">
+          <div className={styles.hiddenDiv}>
```
Comment (mastertheblaster) [inline src/MultiSelect/InputWithTags.js]: @nirhart `div` inside `span` is not good. It should be otherwise.

Comment (mastertheblaster) [inline src/MultiSelect/InputWithTags.js]: @nirhart The div `classname` IMHO should be dynamic, cause the input itself can be small/medium/large.

File: src/MultiSelect/InputWithTags.js [hunk 2]
```diff
@@ -31,10 +33,18 @@ class InputWithTags extends React.Component {
...
+            {this.state.inputValue}
+          </div>
           <Input
             ref={input => this.input = input}
             placeholder={tags.length === 0 ? placeholder : ''}
             {...desiredProps}
+            width="100px"
```
Comment (mastertheblaster) [inline src/MultiSelect/InputWithTags.js]: @nirhart This should be 100% ?

File: src/MultiSelect/InputWithTags.scss [hunk 3]
```diff
@@ -1,7 +1,15 @@
 @import '../common';
 
 .input {
-  width: 100px;
+  min-width: 100px;
+}
+
+.hiddenDiv {
+  @include FontRoman();
+  font-size: 16px;
```
Comment (mastertheblaster) [inline src/MultiSelect/InputWithTags.scss]: @nirhart This should depend on the input size? Or input size is frozen (it's always medium?)

**Comment by mastertheblaster** [comment]
@nirhart The UX issue appears when typing multiple spaces.

**Comment by Faradey27** [comment]
@nirhart maybe better just to use some lib for dynamic input width?
as https://github.com/JedWatson/react-input-autosize or something like this, in such way we will have more general and stable solution

File: src/MultiSelect/InputWithTags.js (hunk 1 above)
Comment (nirhart) [inline src/MultiSelect/InputWithTags.js]: I agree, however the div is invisible so I don't see any problem with that in this case, what do you think?

Comment (nirhart) [inline src/MultiSelect/InputWithTags.js]: medium and large act the same, I added a fix for small

File: src/MultiSelect/InputWithTags.js (hunk 2 above)
Comment (nirhart) [inline src/MultiSelect/InputWithTags.js]: no, it should be 100px, its the initial input size, I don't want it to fit the whole container

File: src/MultiSelect/InputWithTags.scss (hunk 3 above)
Comment (nirhart) [inline src/MultiSelect/InputWithTags.scss]: you are right, I added a fix for it

**Comment by nirhart** [comment]
@Faradey27 I really don't want to do it, we already have an Input of our own, that contains our UX demands properties and our accessibility, I think we should reuse our code in that case

**Comment by nirhart** [comment]
@mastertheblaster fixed the spaces issue

**Comment by mastertheblaster** [comment]
@nirhart seems good

**Pull request merged**